plugins/crypto/openssl/ua_openssl_create_certificate.c

```diff
@@ -0,0 +1,366 @@
+                errRet = UA_STATUSCODE_BADINTERNALERROR;
+                goto cleanup;
+            }
+            outPKey->length = (size_t) tmpLen;
+
+            tmpLen = i2d_X509(x509, &outCert->data);
+            if(tmpLen <= 0) {
+                UA_LOG_ERROR(logger, UA_LOGCATEGORY_SECURECHANNEL,
+                            "Create Certificate: Create DER-certificate failed.");
+                errRet = UA_STATUSCODE_BADINTERNALERROR;
+                goto cleanup;
+            }
+            outCert->length = (size_t) tmpLen;
+            break;
+        }
+        case UA_CERTIFICATE_FORMAT_PEM: {
+            // PKey
+            memPKey = BIO_new(BIO_s_mem());
+            if(!memPKey) {
+                UA_LOG_ERROR(logger, UA_LOGCATEGORY_SECURECHANNEL,
+                            "Create Certificate: Allocate Membuffer for PKey failed.");
+                errRet = UA_STATUSCODE_BADOUTOFMEMORY;
+                goto cleanup;
+            }
+
+            if(PEM_write_bio_PrivateKey(memPKey, pkey, NULL, NULL, 0, 0, NULL) != 1) {
+                UA_LOG_ERROR(logger, UA_LOGCATEGORY_SECURECHANNEL,
+                            "Create Certificate: Generate PEM-PrivateKey failed.");
+                errRet = UA_STATUSCODE_BADINTERNALERROR;
+                goto cleanup;
+            }
+
+            UA_ByteString tmpPem = UA_BYTESTRING_NULL;
+            tmpPem.length = (size_t) BIO_get_mem_data(memPKey, &tmpPem.data);
+            errRet = UA_ByteString_copy(&tmpPem, outPKey);
+            if(errRet != UA_STATUSCODE_GOOD) {
+                UA_LOG_ERROR(logger, UA_LOGCATEGORY_SECURECHANNEL,
+                            "Create Certificate: Copy PEM PKey failed.");
+                goto cleanup;
+            }
+
+            // Cert
+            memCert = BIO_new(BIO_s_mem());
+            if(!memCert) {
+                UA_LOG_ERROR(logger, UA_LOGCATEGORY_SECURECHANNEL,
+                            "Create Certificate: Allocate Membuffer for Cert failed.");
+                errRet = UA_STATUSCODE_BADOUTOFMEMORY;
+                goto cleanup;
+            }
+
+            if(PEM_write_bio_X509(memCert, x509) != 1) {
+                UA_LOG_ERROR(logger, UA_LOGCATEGORY_SECURECHANNEL,
+                            "Create Certificate: Generate PEM-Certifcate failed.");
+                errRet = UA_STATUSCODE_BADINTERNALERROR;
+                goto cleanup;
+            }
+
+            tmpPem.length = (size_t) BIO_get_mem_data(memCert, &tmpPem.data);
+            errRet = UA_ByteString_copy(&tmpPem, outCert);
+            if(errRet != UA_STATUSCODE_GOOD) {
+                UA_LOG_ERROR(logger, UA_LOGCATEGORY_SECURECHANNEL,
+                            "Create Certificate: Copy PEM Certificate failed.");
+                goto cleanup;
+            }
+            break;
+        }
+        default: {
+            UA_LOG_ERROR(logger, UA_LOGCATEGORY_SECURECHANNEL,
```

Comment:
set the return code to something bad.
this could be checked early on in the function.